### PR TITLE
Update zlib from 1.2.6 to 1.2.13

### PR DESCRIPTION
**Pull request by PeterMeisrimelModelon**

### CODE REVIEW

**Review by filip-stenstrom — CHANGES_REQUESTED** [review]
@PeterMeisrimelModelon
Good. Comments are mainly on code style.

Good that you properly documented what we've modified on zlib sources, and updated the version. I think that's enough.

I need to experiment a little with `find_package` myself to get a feel for it. Maybe I'll have more comments after.

File: Config.cmake/fmizlib.cmake [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    SOURCE_DIR  "${ZLIB_SOURCE}"
+    BINARY_DIR  "${FMILIBRARYBUILD}/zlibext"
+    INSTALL_DIR "${ZLIB_INSTALL}"
+    TMP_DIR     "${FMILIBRARYBUILD}/zlibext/temp"
+    STAMP_DIR   "${FMILIBRARYBUILD}/zlibext/stamp"
+)
+
+set(ZLIB_SETTINGS
+    -DBUILD_SHARED_LIBS:BOOLEAN=OFF
+    -DCMAKE_POSITION_INDEPENDENT_CODE:BOOLEAN=ON
```
[the remark this answers is not in the document]
Comment (PeterMeisrimelModelon) [reply Config.cmake/fmizlib.cmake]: Should be, however something within zlib fails to build without, citing the need for `fpic`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -49,8 +49,8 @@ set(FMILIBRARYBUILD ${FMILibrary_BINARY_DIR})
 set(CMAKE_MODULE_PATH ${FMILIBRARYHOME}/Config.cmake)
 
 # User configuration options and parameters
-set(FMILIB_INSTALL_PREFIX ${FMILibrary_BINARY_DIR}/../install CACHE PATH "Prefix prepended to install directories")
-set(FMILIB_THIRDPARTYLIBS  ${FMILibrary_SOURCE_DIR}/ThirdParty CACHE PATH "Path to the ThirdParty library dir" )
+set(FMILIB_INSTALL_PREFIX ${FMILIBRARYBUILD}/../install CACHE PATH "Prefix prepended to install directories")
+set(FMILIB_THIRDPARTYLIBS  ${FMILIBRARYHOME}/ThirdParty CACHE PATH "Path to the ThirdParty library dir" )
```
[the remark this answers is not in the document]
Comment (PeterMeisrimelModelon) [reply CMakeLists.txt]: Renamed to `CMAKE_SOURCE_DIR `and `CMAKE_BINARY_DIR`.

File: Config.cmake/mergestaticlibs.cmake [hunk 3]
```diff
@@ -32,7 +32,7 @@ function(merge_static_libs outlib)
     foreach(lib ${libs})
         get_target_property(libtype ${lib} TYPE)
         if(NOT libtype STREQUAL "STATIC_LIBRARY")
-            message(FATAL_ERROR "Merge_static_libs can only process static\n\tlibraries: ${lib}\n\tlibtype ${libtype}")
+            message(FATAL_ERROR "Merge_static_libs can only process statict libraries\n\tlibraries: ${lib}\n\tlibtype ${libtype}")
```
Comment (filip-stenstrom) [inline Config.cmake/mergestaticlibs.cmake:35]: typo: statict -> static

File: CMakeLists.txt [hunk 4]
```diff
@@ -62,8 +62,8 @@ endif()
 option(FMILIB_BUILD_STATIC_LIB "Build the library as static." ON)
 option(FMILIB_BUILD_SHARED_LIB "Build the library as shared (dll/so/dylib)." ON)
 option(FMILIB_INSTALL_SUBLIBS  "Install sub-libraries (fmicapi. fmixml, jmutil, fmiimport, fmizip)" OFF)
-option(FMILIB_USE_NATIVE_ZLIB  "Use native installation of zlib, possibly requires setting ZLIB_LIBRARY & ZLIB_INCLUDE_DIR to locate installation." OFF)
-mark_as_advanced(FMILIB_USE_NATIVE_ZLIB)
+option(FMILIB_FIND_PACKAGE_ZLIB  "Use find_package for zlib, instead of version in FMILibrary/Thirdparty, possibly requires setting ZLIB_LIBRARY & ZLIB_INCLUDE_DIR." OFF)
```
Comment (filip-stenstrom) [inline CMakeLists.txt:65]: I think it's better to skip the last part with `, possible requires setting ...`. If things stop working when using this option, then a user can read the documentation on ` find_package` and `FindZLIB.cmake` (the errors about missing ZLIB_LIBRARY originates from there).

File: Config.cmake/fmizlib.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (filip-stenstrom) [reply Config.cmake/fmizlib.cmake]: I tested it. fmilib_shared.so which we seem to always build complains. Let's leave this as is.

[resolve]

File: CMakeLists.txt (hunk 4 above)
Comment (PeterMeisrimelModelon) [reply CMakeLists.txt]: Fixed.

**Pull request merged**